catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -1021,27 +999,59 @@ public UpdateResponse update(UpdateRequest updateRequest)
 
         UpdateResponse updateResponse = null;
         try {
+            List<Filter> idFilters = new ArrayList<>();
+            for (Entry<Serializable, Metacard> update : updateReq.getUpdates()) {
+                idFilters.add(filterBuilder.attribute(Metacard.ID)
+                        .is()
+                        .equalTo()
+                        .text(update.getValue()
+                                .getId()));
+            }
+
+            QueryRequestImpl queryRequest = new QueryRequestImpl(
+                    new QueryImpl(filterBuilder.anyOf(idFilters)));
+            QueryResponse query;
+            Map<String, Metacard> metacardMap = new HashMap<>(updateReq.getUpdates()
+                    .size());
+            try {
+                query = doQuery(queryRequest, defaultFederationStrategy);
```

Comment:
I can do that. Just be aware that for the general case, this query will always be executed because we will always have policy/access plugins installed.